internal/provisioner/objects/contourconfig/contourconfig.go

```diff
@@ -88,32 +88,19 @@ func setGatewayConfig(config *contour_api_v1alpha1.ContourConfiguration, contour
 
 // EnsureContourConfigDeleted deletes a ContourConfig for the provided contour, if the configured owner labels exist.
 func EnsureContourConfigDeleted(ctx context.Context, cli client.Client, contour *model.Contour) error {
-	current, err := current(ctx, cli, contour)
-	if err != nil {
-		if errors.IsNotFound(err) {
-			return nil
-		}
-		return err
-	}
-
-	if labels.Exist(current, model.OwnerLabels(contour)) {
-		if err := cli.Delete(ctx, current); err != nil {
-			if errors.IsNotFound(err) {
-				return nil
-			}
-			return err
-		}
+	getter := func(ctx context.Context, cli client.Client, namespace, name string) (client.Object, error) {
+		return current(ctx, cli, namespace, name)
 	}
+	return objects.EnsureObjectDeleted(ctx, cli, contour, contour.ContourConfigurationName(), getter)
```

Comment:
I don't think we need to declare `getter` here, do we? This should work:
```suggestion
	return objects.EnsureObjectDeleted(ctx, cli, contour, contour.ContourConfigurationName(), current)
```

Reply:
no, if don't have it, the compiler is not happy with:
>
> cannot use current (value of type func(ctx context.Context, cli client.Client, namespace string, name string) (*v1alpha1.ContourConfiguration, error)) as type objects.objectGetter in argument to objects.EnsureObjectDeleted
>

Reply:
Ah, must be because the return types are different.